types/src/validator_change.rs

```diff
@@ -35,27 +36,34 @@ impl ValidatorChangeProof {
 
     /// Verify the proof is correctly chained with known epoch and validator verifier
     /// and return the LedgerInfo to start target epoch
+    /// In case waypoint is present it's going to be used for verifying the very first epoch change
+    /// (it's the responsibility of the caller to not pass waypoint in case it's not needed).
```

Comment:
I'd like to split this into two API to be more explicit, one takes a epoch and verifier, another takes waypoint because we only need one of the two but never both